File: src/synapse_agent/tools/jina_reader.py
"""Jina Reader — full-page content fetching."""

from __future__ import annotations

import httpx

from synapse_agent.config import settings


async def fetch_page_content(url: str) -> str:
    """Fetch full page content via Jina Reader API. Returns markdown text."""
    headers: dict[str, str] = {
        "Accept": "application/json",
        "X-Retain-Images": "none",
    }
    if settings.jina_api_key:
        headers["Authorization"] = f"Bearer {settings.jina_api_key}"

    async with httpx.AsyncClient(timeout=30, follow_redirects=True) as client:
        try:
            resp = await client.get(
                f"https://r.jina.ai/{url}",
                headers=headers,
            )
            resp.raise_for_status()
            data = resp.json()
            return data.get("data", {}).get("content", "")
        except Exception:
            return ""

File: src/lib/synapse/tools/jina-reader.ts
/**
 * Jina Reader — full-page content fetching as markdown.
 * Port of Python's tools/jina_reader.py
 */

import { getSettings } from "../config";

export async function fetchPageContent(url: string): Promise<string> {
  const { jinaApiKey } = getSettings();

  const headers: Record<string, string> = {
    Accept: "application/json",
    "X-Retain-Images": "none",
  };
  if (jinaApiKey) {
    headers.Authorization = `Bearer ${jinaApiKey}`;
  }

  try {
    const resp = await fetch(`https://r.jina.ai/${encodeURIComponent(url)}`, {
      headers,
      signal: AbortSignal.timeout(30_000),
    });

    if (!resp.ok) return "";
    const data = await resp.json();
    return (data.data?.content ?? "").slice(0, 15_000);
  } catch {
    return "";
  }
}
